packages/shared/src/components/post/PostActions.tsx

```diff
@@ -77,6 +81,31 @@ export function PostActions({
     await toggleDownvote({ payload: post, origin });
   };
 
+  const [getPost, updatePost] = useUpdateQuery(
+    postByIdQueryOptions({
+      id: post.id,
+    }),
+  );
```

Comment:
Used as a special hook to replace `updatePostCache` and different helpers we have for updating different caches during optimistic actions. This accepts same parameters as `useQuery` so you are sure that updating and getting the data is typed and you don't make accidental mistake.